api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -369,35 +317,82 @@ public ConceptSet toHydratedConceptSet(DbConceptSet dbConceptSet) {
         conceptService.findAll(dbConceptSet.getConceptIds(), CONCEPT_NAME_ORDERING));
   }
 
-  private DbConceptSet fromClientConceptSet(CreateConceptSetRequest request, long workspaceId) {
-    ConceptSet conceptSet = request.getConceptSet();
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    DbConceptSet dbConceptSet = new DbConceptSet();
-    dbConceptSet.setDomainEnum(conceptSet.getDomain());
-    if (conceptSet.getSurvey() != null) {
-      dbConceptSet.setSurveysEnum(conceptSet.getSurvey());
+  private void validateAndUpdateDbConceptSet(DbConceptSet dbConceptSet, ConceptSet conceptSet) {
+    if (Strings.isNullOrEmpty(conceptSet.getEtag())) {
+      throw new BadRequestException("missing required update field 'etag'");
     }
-    if (dbConceptSet.getDomainEnum() == null) {
-      throw new BadRequestException(
-          "Domain " + conceptSet.getDomain() + " is not allowed for concept sets");
+    int version = Etags.toVersion(conceptSet.getEtag());
+    if (dbConceptSet.getVersion() != version) {
+      throw new ConflictException("Attempted to modify outdated concept set version");
     }
-    Optional.ofNullable(conceptSet.getEtag())
-        .ifPresent(etag -> dbConceptSet.setVersion(Etags.toVersion(etag)));
-    dbConceptSet.setDescription(conceptSet.getDescription());
-    dbConceptSet.setName(conceptSet.getName());
-    dbConceptSet.setCreator(userProvider.get());
-    dbConceptSet.setWorkspaceId(workspaceId);
-    dbConceptSet.setCreationTime(now);
-    dbConceptSet.setLastModifiedTime(now);
-    dbConceptSet.setVersion(INITIAL_VERSION);
-    addConceptsToSet(dbConceptSet, request.getAddedIds());
+    if (conceptSet.getName() != null) {
+      dbConceptSet.setName(conceptSet.getName());
+    }
+    if (conceptSet.getDescription() != null) {
+      dbConceptSet.setDescription(conceptSet.getDescription());
+    }
+    if (conceptSet.getDomain() != null && conceptSet.getDomain() != dbConceptSet.getDomainEnum()) {
+      throw new BadRequestException("Cannot modify the domain of an existing concept set");
+    }
+    // In case of rename ConceptDet does not have concepts
+    if (conceptSet.getConcepts() != null) {
```

Comment:
Think we should use `CollectionUtils.isEmpty` here. It does a null and empty check. If the code changes an we passed an empty list during the rename we would still throw a BadRequestException in that case and rename would fail
```suggestion
    if (!CollectionUtils.isEmpty(conceptSet.getConcepts())) {
```